binaryop/funcs.go

```diff
@@ -107,3 +107,21 @@ func Ifnot(left, right float64) float64 {
 	}
 	return nan
 }
+
+// And return left if left and right is not NaN. Otherwise, NaN is returned.
+func And(left, right float64) float64 {
+	if math.IsNaN(left) || math.IsNaN(right) {
+		return nan
+	}
+	return left
+}
+
+// Or return the first non-NaN item. If both left and right are NaN, it returns NaN.
+func Or(left, right float64) float64 {
+	if !math.IsNaN(left) {
+		return left
+	} else if !math.IsNaN(right) {
```

Comment:
It is OK to just return `right` here without checking it for `nan`.

Reply:
Addressed this in the follow-up commit - 363400dae5e18e099a6b8bb87dbedbb0a3be08c6